monitoring/monitoring.go

```diff
@@ -271,8 +281,12 @@ func Start(env e2e.Environment, opts ...Option) (_ *Service, err error) {
 	}
 	env.AddListener(l)
 
-	c := newCadvisor(env, "cadvisor")
-	if err := e2e.StartAndWaitReady(c, p); err != nil {
+	runnables := []e2e.Runnable{p}
+	if opt.useCadvisor {
+		c := newCadvisor(env, "cadvisor")
+		runnables = append(runnables, c)
```

Comment:
so instead of slice, we could just run `e2e.StartAndWaitReady` once for Prometheus and second `e2e.StartAndWaitReady` here (IF option is enabled)